sqldb/sqlc/queries/graph.sql

```diff
@@ -165,13 +165,77 @@ FROM channels c
 WHERE c.scid = $1
   AND c.version = $2;
 
+-- name: GetChannelFeaturesAndExtras :many
+SELECT
+    cf.channel_id,
+    'feature' AS kind,
+    CAST(cf.feature_bit AS TEXT) AS key,
+    NULL AS value
+FROM channel_features cf
+WHERE cf.channel_id = $1
+
+UNION ALL
+
+SELECT
+    cet.channel_id,
+    'extra' AS kind,
+    CAST(cet.type AS TEXT) AS key,
+    cet.value
+FROM channel_extra_types cet
+WHERE cet.channel_id = $1;
+
 -- name: HighestSCID :one
 SELECT scid
 FROM channels
 WHERE version = $1
 ORDER BY scid DESC
 LIMIT 1;
 
+-- name: ListChannelsByNodeID :many
+SELECT c.*,
+    n1.pub_key AS node1_pubkey,
+    n2.pub_key AS node2_pubkey,
+
+    -- Policy 1
+    cp1.id AS policy1_id,
```

Comment:
We could use the new "embed" feature here, which would make the Golang code much simpler I think.
Would look something like this:
```
  SELECT sqlc.embed(c), sqlc.embed(cp1), sqlc.embed(cp2) ...
```

Which will then produce two structs (one for channels and one for channel policies) and re-use them across the different queries.

Reply:
Link: https://docs.sqlc.dev/en/latest/howto/embedding.html (available since a couple of versions).

Reply:
sooooooooo nice thank you 🔥